src/Notify/Notifier/AbstractFcmNotifier.php

```diff
@@ -169,6 +166,30 @@ protected function convertException(\Throwable $exception): array
         ];
     }
 
+    protected function generateAnalyticsLabel(Message $notification): string
+    {
+        return $notification->get('_id')->toString();
+    }
+
+    protected function fetchAccessToken(): string
+    {
+        $client = new \Google_Client();
+        $client->setAuthConfig($this->config);
+        $client->addScope(\Google_Service_FirebaseCloudMessaging::FIREBASE_MESSAGING);
+        $tokens = $client->fetchAccessTokenWithAssertion();
+
+        if (!isset($tokens['access_token'])) {
+            throw new UnableToFetchAccessToken('Failed to retrieve access token.');
+        }
+
+        return $tokens['access_token'];
+    }
+
+    protected function parseConfig(): array
+    {
+        return json_decode(base64_decode(Crypto::decrypt($this->firebaseServiceAccountSecrets, $this->key)), true);
```

Comment:
@bhsiao55 this is an encoded array?  app specific?  seems like this should be in teh app instance/schema.

Reply:
@gdbrown this is originally from the downloaded google service account json file that was downloaded from firebase.  This file is similar to the one being used for CollectArticleStatsHandler, but for this case instead of creating the file, the json of the file has been base64 encoded and then encrypted and being passed in using the aws ssm parameter.  Android, Ios, and the browser will be using this same parameter and is not app specific

Reply:
@efjacobson @bhsiao55 seems like you could just encode everything into the app.  maybe another field for this porpoise.

Reply:
@gdbrown ok I'll create a new field for the fcm apps, currently the fcm_api_key for android is in gdbots
https://github.com/gdbots/schemas/blob/master/schemas/gdbots/iam/mixin/android-app/1-0-3.xml#L16
Is this where this new field should be created as well?

Reply:
@efjacobson @bhsiao55 we can huddle in slack to discuss tomorrow.  i think i get hte gist of what y'all are doing but want to hear the human voice to confirm you're not gpt imposters.